frame/staking/src/lib.rs

```diff
@@ -696,21 +702,45 @@ pub struct IndividualExposure<AccountId, Balance: HasCompact> {
 }
 
 /// A snapshot of the stake backing a single validator in the system.
-#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
-pub struct Exposure<AccountId, Balance: HasCompact> {
+#[derive(
+	PartialEqNoBound,
+	EqNoBound,
+	PartialOrd,
+	Ord,
+	CloneNoBound,
+	Encode,
+	Decode,
+	RuntimeDebugNoBound,
+	TypeInfo,
+	MaxEncodedLen,
+)]
+#[codec(mel_bound())]
+#[scale_info(skip_type_params(MaxNominations))]
+pub struct Exposure<AccountId, Balance, MaxNominations>
+where
+	AccountId: Debug + Encode + Decode + MaxEncodedLen + Clone + Eq + PartialEq,
+	Balance: Debug + HasCompact + Encode + Decode + MaxEncodedLen + Clone + Eq + PartialEq,
+	MaxNominations: Get<u32>,
+{
 	/// The total balance backing this validator.
 	#[codec(compact)]
 	pub total: Balance,
 	/// The validator's own stash that is exposed.
 	#[codec(compact)]
 	pub own: Balance,
 	/// The portions of nominators stashes that are exposed.
-	pub others: Vec<IndividualExposure<AccountId, Balance>>,
+	pub others: BoundedVec<IndividualExposure<AccountId, Balance>, MaxNominations>,
```

Comment:
This is a great direction that you started here: we certainly want to bound this. Problem is that there is nothing in the `ElectionProvider` trait explicitly guarantees that this is the case. 

What we want to do is:

1. Staking tells `ElectionProvider` how many `MaxBackers` it accepts. 
2. Staking miner respects this when generating solution, and automatically sorts the backings of each validator, and throws away the smaller ones. (currently we area doing onchain, which is horrible)
3. `ElectionProvider` checks this for submitted solutions and ensures it is the case.
4.  Staking can just do a defensive truncate when it receives a solution from `ElectionProvider`. 
5. This new `MaxBackers` should also replace `MaxRewardableNominatorsPerValidator`, and entirely replace the notion of oversubscribed validators. 

If interested in doing more steps in this path, I will be happy to do a better writeup about it.